### PR TITLE
feat: [SSCA-1045]: Added InstanceDTO for removal of 126-instance module dependency

### PR DESCRIPTION
**Pull request by arpit-harness**
## Describe your changes

- Added InstanceDTO for replication of Instance Entity
- Added MongoPersistenceTestRule in test suite

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56440)

### CODE REVIEW

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger uts

File: ssca-manager/service/src/main/java/io/harness/changestreams/eventhandlers/InstanceNGRedisEventHandler.java [hunk 1]
```diff
@@ -21,15 +22,16 @@ public class InstanceNGRedisEventHandler extends DebeziumAbstractRedisEventHandl
   @Inject MongoTemplate mongoTemplate;
   @Inject CdInstanceSummaryService cdInstanceSummaryService;
 
-  private Instance createEntity(String value) {
+  @VisibleForTesting
+  public InstanceDTO createEntity(String value) {
     Document document = Document.parse(value);
     document.remove("instanceInfo");
-    return mongoTemplate.getConverter().read(Instance.class, document);
+    return mongoTemplate.getConverter().read(InstanceDTO.class, document);
```
Comment (inderschera) [inline ssca-manager/service/src/main/java/io/harness/changestreams/eventhandlers/InstanceNGRedisEventHandler.java:29]: We should probably use jackson objectMapper to parse json to object.

File: ssca-manager/service/src/test/resources/debezium-create-event-message.txt [hunk 2]
```diff
@@ -1 +1 @@
...
...bWFnZS1uZXc6dGFnLTEifX0sImxhc3REZXBsb3llZEJ5SWQiOiJsdjBldVJoS1JDeWlYV3pTN3BPZzZnIiwibGFzdERlcGxveWVkQnlOYW1lIjoiQWRtaW4iLCJsYXN0RGVwbG95ZWRBdCI6MTY5NTcyODQwMTYzOCwibGFzdFBpcGVsaW5lRXhlY3V0aW9uSWQiOiJqQWZ2TkpNVFIwbWhLMjR0RFp0bUlRIiwibGFzdFBpcGVsaW5lRXhlY3V0aW9uTmFtZSI6IlVsdGltYXRlVGVzdFBpcGVsaW5lIiwic3RhZ2VOb2RlRXhlY3V0aW9uSWQiOiJMZ3ktYXk3eFJtbUtWa2pwbGVRTHNBIiwic3RhZ2VTdGF0dXMiOiJTVUNDRUVERUQiLCJzdGFnZVNldHVwSWQiOiJMZ3ktYXk3eFJtbUtWa2pwbGVRTHNBIiwicm9sbGJhY2tTdGF0dXMiOiJOT1RfU1RBUlRFRCIsImluc3RhbmNlSW5mbyI6eyJuYW1lc3BhY2UiOiJ0ZXN0IiwicmVsZWFzZU5hbWUiOiJyZWxlYXNlLWMxOTI3ODVlNzliOTYyNzJjZDIwNjVhMjcxNWIyNjY0ZGUzYjU5ZTIiLCJwb2ROYW1lIjoiYXJwaXRqaS01YmY2N2JkODY0LWZ6NGZ0IiwicG9kSVAiOiIxMC4wLjAuMTQ2IiwiY29udGFpbmVyTGlzdCI6eyJfMCI6eyJjb250YWluZXJJZCI6ImRvY2tlcjovL2M0MTkwYWIzMGY4ZDRmZDhjYWEzNWQzNGMwZWQ4MDNmOTE0NTJiYjNjNmU5ZWIwOTAzOTc3NGQwYjQ5YTRkMDYiLCJuYW1lIjoiYXJwaXRqaSIsImltYWdlIjoibmdpbng6bGF0ZXN0In19LCJfY2xhc3MiOiJpby5oYXJuZXNzLmVudGl0aWVzLmluc3RhbmNlaW5mby5LOHNJbnN0YW5jZUluZm8ifSwiaXNEZWxldGVkIjpmYWxzZSwiZGVsZXRlZEF0IjoxNjk1NzI5MTkyMDYzLCJjcmVhdGVkQXQiOjE2OTU3Mjg0NzU5ODQsImxhc3RNb2RpZmllZEF0IjoxNjk1NzI4NTkwNzM4LCJfY2xhc3MiOiJpby5oYXJuZXNzLmVudGl0aWVzLkluc3RhbmNlIn0aBkNSRUFURSDTk6f7wzE=
```
[the remark this answers is not in the document]
Comment (arpit-harness) [reply ssca-manager/service/src/test/resources/debezium-create-event-message.txt:1]: Used for unit test. Its an event posted by debezium service.

File: ssca-manager/service/src/main/java/io/harness/changestreams/eventhandlers/InstanceNGRedisEventHandler.java (hunk 1 above)
Comment (arpit-harness) [reply ssca-manager/service/src/main/java/io/harness/changestreams/eventhandlers/InstanceNGRedisEventHandler.java:29]: Used mongotemplate as this was an operation equivalent to reading the mongo data into an entity class.

**Comment by arpit-harness** [comment]
trigger uts

**Comment by arpit-harness** [comment]
trigger ut5 ut6 ut7 ut8 ut9 ut2

**Review by inderschera — APPROVED** [review]

**Pull request merged**